### DISCUSSION TITLE
How triggers work when performing bulk operation?

### DISCUSSION THREAD

**Discussion by MacyF** [koenbeuk/EntityFrameworkCore.Triggered#153] [Q&A] [answered]
I'm wondering if a trigger get fired n times for n entities that gets updated in bulk.

For example, I have an application that notifies students of their status (Approved or Not) when their grade are uploaded. And he grades are uploaded all at once for all the students (bulk operation).

So, what I'm trying to ask is that the trigger get fired for each student in the bulk or a single time for the bulk operation itself.

Thanks in advance.

**Comment by koenbeuk** [comment, marked as answer]
A trigger will always fire for each individual changed entity in the ChangeTracker. So yes, A trigger will fire N times for N entities.

**Reply by MacyF** [reply]
Ok, thank for your answer.
I'm wondering to implement triggers and was concerned about not being able to track all entity in a bulk operation and perform modification accordingly.
